src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -525,6 +527,21 @@ private boolean isContainVerifiableType(DetailAST modifiers) {
         return result;
     }
 
+    /**
+     * Checks a record component definition. Record components are private
+     * final fields, so we check code in a similar manner as we do for
+     * private final class members.
+     *
+     * @param recordComponentDef the record component definition to check.
+     */
+    private void visitRecordComponentDef(DetailAST recordComponentDef) {
+        if (memberModifiers.isEmpty()
+                || memberModifiers.contains(TokenTypes.LITERAL_PRIVATE)
+                || memberModifiers.contains(TokenTypes.FINAL)) {
```

Comment:
I do not think this conditions are required.
See my comment above that it should be validated  like ctor

Reply:
@romani sir, should I push my previous changes CONDITIONS again instead of this?
Please checkout: https://github.com/Rahulkhinchi03/checkstyle/commit/7731e7cfa2c300bb36d4ba13f233075ca230b9e1

Reply:
Done!

Reply:
```
case TokenTypes.RECORD_COMPONENT_DEF:
                visitComponentDef(ast);
                break;
```

```
  /**
     * Checks type of ComponentDef.
     *
     * @param variableDef to check.
     */
    private void visitComponentDef(DetailAST variableDef) {
        if (isContainVerifiableType(variableDef)) {
            checkClassName(variableDef);
        }
    }
```
Using this code , I am getting violation error in file ` InputIllegalTypeRecordsAndCompactCtor`,



1. Should I use this code only and resolve these errors too?
2. Or Should I use: 
 ```
private void visitRecordComponentDef(DetailAST recordComponentDef) {
        if (memberModifiers.isEmpty()) {
            checkClassName(recordComponentDef);
        }
    }
```

Using point 2, I am getting no error.

Reply:
see comment at https://github.com/checkstyle/checkstyle/pull/10958#discussion_r762465306